From 0484361d765214c98191eb5486008a66b02c746c Mon Sep 17 00:00:00 2001
From: Y300-0100 <bosko.brajovich@gmail.com>
Date: Thu, 4 Sep 2014 10:33:04 +0200
Subject: [PATCH 1/2] Update main.cpp

---
 main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main.cpp b/main.cpp
index 5197b35..e6a99ec 100644
--- a/main.cpp
+++ b/main.cpp
@@ -189,7 +189,7 @@ static int process_config(VolumeManager *vm)
                 flags |= VOL_PROVIDES_ASEC;
             }
             /* Set this flag if the storage device supports having external apps */
-            if (strcmp(fstab->recs[i].label, EXTERNAL_APPS_DEVICE_LABEL) == 0) {
+            if (strcmp(fstab->recs[i].label, EXTERNAL_APPS_DEVICE_LABEL) == 1) {
                 flags |= VOL_EXTERNAL_APPS;
             }
             dv = new DirectVolume(vm, &(fstab->recs[i]), flags);
-- 
1.9.1
