From 802f86c20187d0b98ad99158efdf89afbe9a0d97 Mon Sep 17 00:00:00 2001
From: Y300-0100 <bosko.brajovich@gmail.com>
Date: Fri, 8 Aug 2014 16:05:24 +0200
Subject: [PATCH 2/3] Update AwesomePlayer.cpp

---
 media/libstagefright/AwesomePlayer.cpp |   38 +++++++++-----------------------
 1 file changed, 11 insertions(+), 27 deletions(-)

diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index 6bbf8b4..9a34169 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -77,7 +77,11 @@
 
 #include <cutils/properties.h>
 
+#ifdef QCOM_LEGACY_OMX
+#define USE_SURFACE_ALLOC 0
+#else
 #define USE_SURFACE_ALLOC 1
+#endif
 #define FRAME_DROP_FREQ 0
 #ifdef QCOM_HARDWARE
 #define LPA_MIN_DURATION_USEC_ALLOWED 30000000
@@ -804,7 +808,7 @@ void AwesomePlayer::ensureCacheIsFetching_l() {
 
 void AwesomePlayer::onVideoLagUpdate() {
     Mutex::Autolock autoLock(mLock);
-    if (!mVideoLagEventPending || mAudioPlayer == NULL) {
+    if (!mVideoLagEventPending) {
         return;
     }
     mVideoLagEventPending = false;
@@ -1080,9 +1084,6 @@ status_t AwesomePlayer::play_l() {
             }
 
             if (err != OK) {
-                mAudioSource.clear();
-                mOmxSource.clear();
-
                 delete mAudioPlayer;
                 mAudioPlayer = NULL;
 
@@ -1156,17 +1157,15 @@ status_t AwesomePlayer::fallbackToSWDecoder() {
     if (!(mFlags & AUDIOPLAYER_STARTED)) {
         mAudioSource->stop();
     }
-    mAudioSource.clear();
     modifyFlags((AUDIO_RUNNING | AUDIOPLAYER_STARTED), CLEAR);
     mOffloadAudio = false;
-#ifdef ENABLE_AV_ENHANCEMENTS
-    // no 24-bit for fallback
-    ExtendedUtils::updateOutputBitWidth(mAudioSource->getFormat(), false);
-#endif
-
     mAudioSource = mOmxSource;
     if (mAudioSource != NULL) {
-        if ((err = mAudioSource->start()) == OK) {
+        err = mAudioSource->start();
+
+        if (err != OK) {
+            mAudioSource.clear();
+        } else {
             mSeekNotificationSent = true;
             if (mExtractorFlags & MediaExtractor::CAN_SEEK) {
                 seekTo_l(curTimeUs);
@@ -2337,9 +2336,6 @@ void AwesomePlayer::onVideoEvent() {
             ALOGE("Failed to fallback to SW decoder err = %d", err);
             notifyListener_l(MEDIA_ERROR, MEDIA_ERROR_UNKNOWN, err);
 
-            mAudioSource.clear();
-            mOmxSource.clear();
-
             delete mAudioPlayer;
             mAudioPlayer = NULL;
 
@@ -3426,20 +3422,8 @@ bool AwesomePlayer::isWidevineContent() const {
 
 status_t AwesomePlayer::dump(int fd, const Vector<String16> &args) const {
     Mutex::Autolock autoLock(mStatsLock);
-    int dfd = dup(fd);
 
-    if (dfd < 0) {
-        ALOGE("dump: failed to dup file descriptor");
-        return -errno;
-    }
-
-    FILE *out = fdopen(dfd, "w");
-
-    if (!out) {
-        ALOGE("dump: failed to open file");
-        close(dfd);
-        return -ENOMEM;
-    }
+    FILE *out = fdopen(dup(fd), "w");
 
     fprintf(out, " AwesomePlayer\n");
     if (mStats.mFd < 0) {
-- 
1.7.9.5
