From 93935281220cc42629ab896203aa98de8144d5cd Mon Sep 17 00:00:00 2001
From: Y300-0100 <bosko.brajovich@gmail.com>
Date: Fri, 8 Aug 2014 16:07:35 +0200
Subject: [PATCH 3/3] Update Utils.cpp

---
 media/libstagefright/Utils.cpp |    9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/media/libstagefright/Utils.cpp b/media/libstagefright/Utils.cpp
index 551af0e..9895ad2 100644
--- a/media/libstagefright/Utils.cpp
+++ b/media/libstagefright/Utils.cpp
@@ -40,7 +40,6 @@
 #ifdef ENABLE_AV_ENHANCEMENTS
 #include <QCMediaDefs.h>
 #include <QCMetaData.h>
-#include "include/ExtendedUtils.h"
 #endif
 
 namespace android {
@@ -675,13 +674,7 @@ bool canOffloadStream(const sp<MetaData>& meta, bool hasVideo, const sp<MetaData
 
     // Check if offload is possible for given format, stream type, sample rate,
     // bit rate, duration, video and streaming
-    bool canOffload = AudioSystem::isOffloadSupported(info);
-
-#ifdef ENABLE_AV_ENHANCEMENTS
-    ExtendedUtils::updateOutputBitWidth(meta, canOffload);
-#endif
-
-    return canOffload;
+    return AudioSystem::isOffloadSupported(info);
 }
 
 void printFileName(int fd)
-- 
1.7.9.5
