From 516e48dce096e2d1ec322f498f1f1489b48b7fde Mon Sep 17 00:00:00 2001
From: Y300-0100 <bosko.brajovich@gmail.com>
Date: Wed, 30 Jul 2014 16:43:11 +0200
Subject: [PATCH 1/2] Remove SHARED_LIBRARIES libqdMetaData
---
 libs/gui/Android.mk |    1 -
 1 file changed, 1 deletion(-)

diff --git a/libs/gui/Android.mk b/libs/gui/Android.mk
index ca78f86..4f12102 100644
--- a/libs/gui/Android.mk
+++ b/libs/gui/Android.mk
@@ -55,7 +55,6 @@ else
 endif
     LOCAL_C_INCLUDES        += $(TARGET_OUT_HEADERS)/vpu/
     LOCAL_CFLAGS            += -DQCOM_BSP
-    LOCAL_SHARED_LIBRARIES  += libqdMetaData
 endif
 
 ifeq ($(BOARD_EGL_SKIP_FIRST_DEQUEUE),true)
-- 
1.7.9.5
