From 96fb1bdf8cf83997eecc93938ca6b71b240808a1 Mon Sep 17 00:00:00 2001
From: Y300-0100 <bosko.brajovich@gmail.com>
Date: Wed, 30 Jul 2014 19:23:09 +0200
Subject: [PATCH] Track Y300 repos

---
 tools/roomservice.py |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tools/roomservice.py b/tools/roomservice.py
index 4c40a05..d88d9ef 100755
--- a/tools/roomservice.py
+++ b/tools/roomservice.py
@@ -34,16 +34,16 @@ except ImportError:
 
 # Config
 # set this to the default remote to use in repo
-default_rem = "lsd"
+default_rem = "gh"
 # set this to the default revision to use (branch/tag name)
-default_rev = "kitkat"
+default_rev = "cm-11.0"
 # set this to the remote that you use for projects from your team repos
 # example fetch="https://github.com/omnirom"
 default_team_rem = "gh"
 # this shouldn't change unless google makes changes
 local_manifest_dir = ".repo/local_manifests"
 # change this to your name on github (or equivalent hosting)
-android_team = "LiquidSmooth-Devices"
+android_team = "Y300-0100"
 
 
 def check_repo_exists(git_data):
@@ -200,7 +200,7 @@ def parse_device_from_folder(device):
     elif len(search) == 1:
         location = search[0]
     else:
-        print("Your device was not found. Attempting to retrieve device repository from LiquidSmooth Github..")
+        print("Your device was not found. Attempting to retrieve device repository from Y300-0100 Github..")
         location = parse_device_from_manifest(device)
     return location
 
-- 
1.7.9.5
