From 5be5cd726cfc53ef85b6354b70f39208b984556b Mon Sep 17 00:00:00 2001
From: Y300-0100 <bosko.brajovich@gmail.com>
Date: Mon, 4 Aug 2014 16:05:18 +0200
Subject: [PATCH 7/7] Update TARGET_linux-x86.mk

---
 core/combo/TARGET_linux-x86.mk |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/combo/TARGET_linux-x86.mk b/core/combo/TARGET_linux-x86.mk
index 69a8035..1a7154f 100644
--- a/core/combo/TARGET_linux-x86.mk
+++ b/core/combo/TARGET_linux-x86.mk
@@ -65,7 +65,7 @@ TARGET_LIBGCC := \
 	$(shell $(TARGET_CC) -m32 -print-file-name=libgcc.a)
 endif
 
-TARGET_NO_UNDEFINED_LDFLAGS := -Wl,--no-undefined
+TARGET_NO_UNDEFINED_LDFLAGS := -W
 
 libc_root := bionic/libc
 libm_root := bionic/libm
-- 
1.7.9.5
