From 63bf200c7ad81adf3c0827140d9bf8c714cafe0b Mon Sep 17 00:00:00 2001
From: Y300-0100 <bosko.brajovich@gmail.com>
Date: Wed, 30 Jul 2014 19:11:10 +0200
Subject: [PATCH] Update vendorsetup.sh

---
 vendorsetup.sh |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vendorsetup.sh b/vendorsetup.sh
index 29d24f1..11210ae 100644
--- a/vendorsetup.sh
+++ b/vendorsetup.sh
@@ -62,4 +62,4 @@ add_lunch_combo liquid_umts_spyder-userdebug
 add_lunch_combo liquid_v500-userdebug
 add_lunch_combo liquid_vigor-userdebug
 add_lunch_combo liquid_yuga-userdebug
-add_lunch_combo liquid_yuga-userdebug
+add_lunch_combo liquid-u8833-userdebug
-- 
1.7.9.5
